File: trunk/Projects/Framework-Samples/PLINQO/CSharp/Tracker/Source/Tracker.Memcached.Tests/CacheTests/ClearCacheTests.cs
using System.Linq;
using System.Web;
using CodeSmith.Data.Linq;
using NUnit.Framework;
using Tracker.Core.Data;

namespace Tracker.Memcached.Tests.CacheTests
{
    [TestFixture]
    public class ClearCacheTests : MemcachedTestBase
    {
        [Test]
        public void SimpleTest()
        {
            try
            {
                using (var db = new TrackerDataContext())
                {
                    var query = db.Role.Where(r => r.Name == "Test Role");
                    var key = query.GetHashKey();
                    var roles = query.FromCache();

                    var cache1 = HttpRuntime.Cache.Get(key);
                    Assert.IsNotNull(cache1);
                    Assert.AreSame(roles, cache1);

                    query.ClearCache();

                    var cache2 = HttpRuntime.Cache.Get(key);
                    Assert.IsNull(cache2);
                }
            }
            catch (AssertionException)
            {
                throw;
            }
            catch
            {
                Assert.Fail();
            }
        }
    }
}

File: trunk/Projects/Framework-Samples/PLINQO/VisualBasic/Tracker/Source/Tracker.Tests/CacheTests/ClearCacheTests.vb
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports CodeSmith.Data.Linq
Imports NUnit.Framework
Imports Tracker.Core.Data

Namespace Tracker.Tests.CacheTests
    <TestFixture()> _
    Public Class ClearCacheTests
        Inherits RoleTests
        <Test()> _
        Public Sub SimpleTest()
            Using db As New TrackerDataContext() With { _
             .Log = Console.Out _
            }
                Dim query As IQueryable(Of Role) = db.Role.Where(Function(r) r.Name = "Test Role")
                Dim key As String = query.GetHashKey()
                Dim roles As IEnumerable(Of Role) = query.FromCache()

                Dim cache1 As Object = HttpRuntime.Cache.[Get](key)
                Assert.IsNotNull(cache1)

                query.ClearCache()

                Dim cache2 As Object = HttpRuntime.Cache.[Get](key)
                Assert.IsNull(cache2)
            End Using
        End Sub
    End Class
End Namespace
